File: MenuSysStack.cpp
#include<bits/stdc++.h>
using namespace std;

class Stack
    {
    public:
        vector<int>x;

        void push(int value)
        {
            x.push_back(value);
        }

        void pop()
        {
            if(x.size()>0) x.pop_back();
            else cout<<"Stack is empty\n";
        }

        void top()
        {
            if(x.size()>0) cout<<"Top Element : "<<x.back()<<endl;
            else cout<<"Stack is empty\n";
        }
        bool empty()
        {
            return x.size()==0;
        }
        int size()
        {
            return x.size();
        }

        void insert(int index,int value)
        {
            if(index<0 || index>x.size())
            {
                cout<<"Inavlid Index\n";
                return;
            }
            x.insert(x.begin()+index,value);
            cout<<"Inserted...!\n";
        }

        void deleteat(int index)
        {
            if(index<0 || index>=x.size())
            {
                cout<<"Invalid Index !\n";
                return ;
            }
            x.erase(x.begin()+index);
            cout<<"Deleted......!\n";
        }
        void show()
        {
            if(x.empty())
            {
                cout<<"Stack is empty\n";
                return;
            }
            cout<<"Stack element : ";
            for(int v:x) cout<<v<<" ";
            cout<<endl;
        }
    };

    int main()
    {
        Stack s;
        int ch;

        while(ch)
        {
            cout<<"\n-----MENU-----\n";
            cout<<"1. Push\n";
            cout<<"2. Pop\n";
            cout<<"3. Top Show\n";
            cout<<"4. Insert Element\n";
            cout<<"5. Delete Element\n";
            cout<<"6. Show All Element\n";
            cout<<"7. Size Element\n";
            cout<<"8. Exit\n";
            cout<<"Enter your choice : ";
            cin>>ch;

            if(ch==1)
            {
                int v;
                cout<<"Enter Value : ";
                cin>>v;
                s.push(v);
            }
            else if(ch==2)
            {
                s.pop();
            }
            else if(ch==3)
            {
                s.top();
            }
            else if(ch==4)
            {
                int index,value;
                cout<<"Enter index : ";
                cin>>index;
                cout<<"Enter Value : ";
                cin>>value;
                s.insert(index,value);
            }
            else if(ch==5)
            {
                int index;
                cout<<"Enter index : ";
                cin>>index;
                s.deleteat(index);
            }
            else if(ch==6)
            {
                s.show();
            }
            else if(ch==7)
            {
                cout<<"Element Size : "<<s.size();
            }
            else if(ch==8)
            {
                cout<<"Exiting...!\n";
                break;
            }
            else
            {
                cout<<"Invalid Choice..!\n";
            }
        }
        return 0;
    }

File: MenuSysStack.py
class Stack:
    def __init__(self):
        self.x = []

    def push(self, value):
        self.x.append(value)

    def pop(self):
        if len(self.x) > 0:
            self.x.pop()
        else:
            print("Stack is empty")

    def top(self):
        if len(self.x) > 0:
            print("Top Element :", self.x[-1])
        else:
            print("Stack is empty")

    def empty(self):
        return len(self.x) == 0

    def size(self):
        return len(self.x)

    def insert(self, index, value):
        if index < 0 or index > len(self.x):
            print("Invalid Index")
            return
        self.x.insert(index, value)
        print("Inserted.....!")

    def deleteat(self, index):
        if index < 0 or index >= len(self.x):
            print("Invalid Index.")
            return
        self.x.pop(index)
        print("Deleted...!")

    def show(self):
        if len(self.x) == 0:
            print("Stack is empty")
            return
        print("Stack Element :", *self.x)


# Main Program
s = Stack()

while True:
    print("\n-----MENU-----")
    print("1. Push")
    print("2. Pop")
    print("3. Top Show")
    print("4. Insert Element")
    print("5. Delete Element")
    print("6. Show All Element")
    print("7. Size Element")
    print("8. Exit")

    ch = int(input("Enter your choice : "))

    if ch == 1:
        value = int(input("Enter Value : "))
        s.push(value)

    elif ch == 2:
        s.pop()

    elif ch == 3:
        s.top()

    elif ch == 4:
        index = int(input("Enter Index : "))
        value = int(input("Enter Value : "))
        s.insert(index, value)

    elif ch == 5:
        index = int(input("Enter Index : "))
        s.deleteat(index)

    elif ch == 6:
        s.show()

    elif ch == 7:
        print("Element Size :", s.size())

    elif ch == 8:
        print("Exiting...")
        break

    else:
        print("Invalid choice!")
